File: tj-aigc/src/main/java/com/tianji/aigc/agent/BuyAgent.java
package com.tianji.aigc.agent;

import cn.hutool.core.map.MapUtil;
import com.tianji.aigc.config.SystemPromptConfig;
import com.tianji.aigc.constants.Constant;
import com.tianji.aigc.enums.AgentTypeEnum;
import com.tianji.aigc.tools.OrderTools;
import com.tianji.common.utils.UserContext;
import lombok.RequiredArgsConstructor;

import org.springframework.ai.vectorstore.VectorStore;
import org.springframework.stereotype.Component;


import java.util.Map;

@Component
@RequiredArgsConstructor
public class BuyAgent extends AbstractAgent{

    private final SystemPromptConfig systemPromptConfig;
    private final OrderTools orderTools;
    private final VectorStore vectorStore;


    @Override
    public AgentTypeEnum getAgentType() {
        return AgentTypeEnum.BUY;
    }

    @Override
    public String systemMessage() {
        return this.systemPromptConfig.getBuyAgentSystemMessage().get();
    }

    @Override
    public Object[] tools() {
        return new Object[]{orderTools};
    }

    @Override
    public Map<String, Object> toolContext(String sessionId, String requestId) {
        var userId = UserContext.getUser();
        return MapUtil.<String, Object>builder()
                .put(Constant.USER_ID, userId)
                .put(Constant.REQUEST_ID, requestId)
                .build();
    }


}

File: tj-aigc-py/app/agents/buy_agent.py
from app.agents.base import AbstractAgent
from app.core.prompts import PromptKey
from app.schemas.enums import AgentTypeEnum
from app.tools.order_tools import ORDER_TOOLS


class BuyAgent(AbstractAgent):

    def get_agent_type(self) -> AgentTypeEnum:
        return AgentTypeEnum.BUY

    @property
    def prompt_key(self) -> PromptKey:
        return PromptKey.BUY_AGENT

    def tools(self) -> list[dict]:
        return ORDER_TOOLS
